pkg/server/local_observer.go

```diff
@@ -220,12 +220,20 @@ func (s *LocalObserverServer) GetFlows(
 	req *observer.GetFlowsRequest,
 	server observer.Observer_GetFlowsServer,
 ) (err error) {
+	ctx := server.Context()
```

Comment:
I think we should actually wrap this context here in a `WithCancel` and call cancel when `GetFlows` returns, to ensure that the context lives only as as long as the request itself. This allows implementors of the `OnGetFlows` hook to release any resources it allocated for this particular request.

We probably also want to pass this per-request-context down to the `OnBuildFilter` and `OnFlowDelivery` hooks as well.

Reply:
Update: I have fixed the context in #216.